numpyro/infer/hmc.py

```diff
@@ -252,6 +253,9 @@ def init_kernel(
             `d2` is the max tree depth during post warmup phase.
         :param bool find_heuristic_step_size: whether to a heuristic function to adjust the
             step size at the beginning of each adaptation window. Defaults to False.
+        :param bool regularize_mass_matrix: whether or not to regularize the estimated mass
```

Comment:
do you think it'd be more useful to set the regularization scale instead? (so e.g. `0.0` corresponds to no regularization?)

Reply:
Currently, the regularization does
```python
var = var * (n / (n + 5)) + 1e-3 * (5 / (n + 5))
```
There are a few numbers 5, 1e-3, n. Do you suggest
```
var = var + scale * (1e-3 - var) * (5 / (n + 5))
```
so that `scale=0` means no regularization, `scale=1` means the old regularization? I feel like this regularization is not needed and might only be useful for dense mass in early warmup phase. So a boolean flag here is enough I guess.

Reply:
hmm i meant the `1e-3` term but i guess setting that to zero doesn't mean there's no regularization... maybe there are too many parameters to expose meaningfully